pkgs/os-specific/darwin/install-binary-package/unpack-dmg-pkg.sh

```diff
@@ -0,0 +1,31 @@
+# shellcheck shell=bash
+unpackCmdHooks+=(unpackDmgPkg)
+unpackDmgPkg() {
+    case "$curSrc" in
+        *.dmg) unpackDmg "$curSrc";;
+        *.pkg) unpackPkg "$curSrc";;
+        *) return 1
+    esac
+}
+
+unpackDmg() {
+    _unpack "$1" "${2:-$pname}"
+}
+
+unpackPkg() {
+    _unpack "$1" "${2:-$pname}"
+    pushd "${2:-$pname}"
+    # Extract files from Payload or Payload~ file contained in given pkg if any
+    # Depending on how apps are packaged maxdepth might need to be adjusted
+    # or made adjustable
+    find . -name 'Payload*' -maxdepth 2 -print0 | xargs -0 -t -I % bsdtar -xf %
+    popd
```

Comment:
I kind of would like to use a sub shell with `cd ...` here to avoid leaking a directory change in case of errors`